osquery/runtime.go

```diff
@@ -565,10 +570,12 @@ func (r *Runner) launchOsqueryInstance() error {
 // instance with the same configurations.
 func (r *Runner) Restart() error {
 	r.instanceLock.Lock()
-	defer r.instanceLock.Unlock()
 	// Cancelling will cause all of the cleanup routines to execute, and a
 	// new instance will start.
 	r.instance.cancel()
+	r.instance.errgroup.Wait()
+	r.instanceLock.Unlock()
```

Comment:
Why switch this from defer?

Reply:
tests weren't passing initially, but maybe this is not needed anymore. Let me rerun the test with a defer.